instrumentation/shoryuken/lib/opentelemetry/instrumentation/shoryuken/middlewares/server/tracer_middleware.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+module OpenTelemetry
+  module Instrumentation
+    module Shoryuken
+      module Middlewares
+        module Server
+          # TracerMiddleware propagates context and instruments Shoryuken requests
+          # by way of its middleware system
+          class TracerMiddleware
+            def call(worker_instance, queue, sqs_msg, _body) # rubocop:disable Metrics/MethodLength
+              attributes = {
+                OpenTelemetry::SemanticConventions::Trace::MESSAGING_SYSTEM => 'shoryuken',
+                'messaging.shoryuken.job_class' => worker_instance.class.name,
+                OpenTelemetry::SemanticConventions::Trace::MESSAGING_MESSAGE_ID => sqs_msg.message_id,
+                OpenTelemetry::SemanticConventions::Trace::MESSAGING_DESTINATION => queue,
+                OpenTelemetry::SemanticConventions::Trace::MESSAGING_DESTINATION_KIND => 'queue',
+                OpenTelemetry::SemanticConventions::Trace::MESSAGING_OPERATION => 'process'
+              }
+              if sqs_msg.respond_to?(:attributes)
+                sqs_msg.attributes.each do |k, v|
+                  attributes["messaging.attributes.#{k}"] = v
+                end
+              end
+              span_name = case instrumentation_config[:span_naming]
+                          when :job_class then "#{worker_instance.class.name} process"
+                          else "#{queue} process"
+                          end
+
+              extracted_context = OpenTelemetry.propagation.extract(sqs_msg)
+              OpenTelemetry::Context.with_current(extracted_context) do
+                links = []
+                span_context = OpenTelemetry::Trace.current_span(extracted_context).context
+                links << OpenTelemetry::Trace::Link.new(span_context) if span_context.valid?
+                span = tracer.start_root_span(span_name, attributes: attributes, links: links, kind: :consumer)
+                OpenTelemetry::Trace.with_span(span) do
+                  yield
+                rescue Exception => e # rubocop:disable Lint/RescueException
+                  span.record_exception(e)
```

Comment:
Something to consider here is that for some reason an error occurs allocating the span this will result in a `NoMethodError` on `nil`.

Please consider using safe navigation for span operations.